### ISSUE TITLE
undefined is not an object (evaluating '_react2.PropTypes.arrayOff')

### ISSUE THREAD

**Issue by thanhppm** [haqiqiw/react-native-slideshow#4]
Hello, when i have devlopment with react-native-slideshow, i have one off issue be like 
"undefined is not an object (evaluating '_react2.PropTypes.arrayOff')". 
this is errors display http://prntscr.com/hhm4u1
Could you help me? Thanks

**Comment by becnichelesmith** [comment]
I have the same issue

**Comment by IElgohary** [comment]
It's because in later versions of react, prop-types is an independent npm module.
just run
`npm install --save prop-types`
and in Slideshow.js, repove PropTypes from the imports from React and add this line:
`import PropTypes from 'prop-types';`

**Comment by thanhppm** [comment]
Thank you @IElgohary

**Comment by adityajhordan** [comment]
@IElgohary yup, it solved, but every time I did rm -rf node_modules then npm install, we have to write PropTypes at SlideShow.js again, is there any solution for that ?

**Comment by hesusdios** [comment]
@IElgohary What do you mean? I added the npm install and imported proptypes but i still get an error.

**Comment by hesusdios** [comment]
import React  from 'react';
import { View, Text, Button, StyleSheet } from 'react-native';
import Slideshow from 'react-native-slideshow';
import PropTypes from 'prop-types';




class SlideshowGallery extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            position: 1,
            interval: null,
            dataSource: [
                {
                    title: 'Title 1',
                    caption: 'Caption 1',
                    url: 'http://placeimg.com/640/480/any',
                }, {
                    title: 'Title 2',
                    caption: 'Caption 2',
                    url: 'http://placeimg.com/640/480/any',
                }, {
                    title: 'Title 3',
                    caption: 'Caption 3',
                    url: 'http://placeimg.com/640/480/any',
                },
            ],
        };
    }

    componentWillMount() {
        this.setState({
            interval: setInterval(() => {
                this.setState({
                    position: this.state.position === this.state.dataSource.length ? 0 : this.state.position + 1
                });
            }, 2000)
        });
    }

    componentWillUnmount() {
        clearInterval(this.state.interval);
    }

    render() {
        return (
            <Slideshow
                dataSource={this.state.dataSource}
                position={this.state.position}
                onPositionChanged={position => this.setState({ position })} />
        );
    }
}

export default SlideshowGallery;

**Comment by wmars91** [comment]
@hesusdios import your prop-type at node_modules/react-native-slideshow/Slideshow.js

**Comment by ahmedcs1995** [comment]
It works 👍 
npm install --save prop-types
and in Slideshow.js, remove PropTypes from the imports from React and add this line:
import PropTypes from 'prop-types';

**Comment by IElgohary** [comment]
@adityajhordan  Sorry for the late reply.
Personally, I just took the code from Slideshow.js and added it to my code.

**Comment by ga-reth** [comment]
Hi, I have used the above solutions:
+npm install --save prop-types
+Slideshow.js: import PropTypes from 'prop-types';

However, I'm still receiving the error.

**Comment by marcmoo** [comment]
haqihaqi, could you patch your code as IElgohary suggested?

**Comment by Muedgar** [comment]
I am using react-native-slideshow but I get this error:
Cannot read property 'arrayOf' of undefined.

I did as some answer suggested:
// install prop-types module
// then import it in the codebase
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Slideshow from 'react-native-slideshow';

export default class App extends Component {

render() {
return (
<Slideshow
dataSource={[
{ url:'http://placeimg.com/640/480/any' },
{ url:'http://placeimg.com/640/480/any' },
{ url:'http://placeimg.com/640/480/any' }
]}/>
);
}
}

But I am still getting the same error:
Help me please

**Comment by Muedgar** [comment]
I solved the issue

**Comment by robiadiprawira** [comment]
@Muedgar how u solved the issue ? 

i have same issue here ...

**Comment by mercxry** [comment]
@haqiqiw Can you push this fix to npm?

**Comment by SIR-MHD** [comment]
Hi everybody.
I have the same issue, i tried solutions up here and still have the error 
how do i fix this, please help

**Comment by suresssh** [comment]
it can be fixed by removing Proptypes which import from react, then  import Proptypes from prop-types in Slideshow.js (react-native-slideshow).

**Comment by syyam** [comment]
@Muedgar  How did you solve that?

**Comment by Uzairrrr** [comment]
npm install --save prop-types
and in Slideshow.js,(node module/react-native-slideshow/slideshow.js) remove PropTypes from the imports from React and add this line:
import PropTypes from 'prop-types';

**Comment by salmansayyidulhaq** [comment]
> import React from 'react';
> import { View, Text, Button, StyleSheet } from 'react-native';
> import Slideshow from 'react-native-slideshow';
> import PropTypes from 'prop-types';
> 
> class SlideshowGallery extends React.Component {
> constructor(props) {
> super(props);
> 
> ```
>     this.state = {
>         position: 1,
>         interval: null,
>         dataSource: [
>             {
>                 title: 'Title 1',
>                 caption: 'Caption 1',
>                 url: 'http://placeimg.com/640/480/any',
>             }, {
>                 title: 'Title 2',
>                 caption: 'Caption 2',
>                 url: 'http://placeimg.com/640/480/any',
>             }, {
>                 title: 'Title 3',
>                 caption: 'Caption 3',
>                 url: 'http://placeimg.com/640/480/any',
>             },
>         ],
>     };
> }
> 
> componentWillMount() {
>     this.setState({
>         interval: setInterval(() => {
>             this.setState({
>                 position: this.state.position === this.state.dataSource.length ? 0 : this.state.position + 1
>             });
>         }, 2000)
>     });
> }
> 
> componentWillUnmount() {
>     clearInterval(this.state.interval);
> }
> 
> render() {
>     return (
>         <Slideshow
>             dataSource={this.state.dataSource}
>             position={this.state.position}
>             onPositionChanged={position => this.setState({ position })} />
>     );
> }
> ```
> }
> 
> export default SlideshowGallery;

can it be from firebase?
